### PR TITLE
Python logging, error handling and imports

### PR DESCRIPTION
**Pull request by loiccoyle**
As discussed with @luxiaohan this PR replaces utils.message with python's standard logging library ["logging"](https://docs.python.org/3/library/logging.html). This means that we no longer have to carry around the mes_level and log_file variables throughout pysixdesk. Instead the user can set the logging settings for the entire module, I show some examples in comments in the __init__.py, perhaps these should go in the README.md ? 
One potential drawback is for the jobs running on condor, they may not respect the users logginglevel/logfile choice in this current implementation but, I'm sure it can be done, if necessary.

I also renamed the lib folder to pysixdesk and removed the highest level __init__.py as I assume you don't want the files LICENSE, CERN-logo.png,... to be part of python package. This means the imports required to run a study,  now look like:`from pysixdesk import WorkSpace`. And the PYTHONPATH shell variable also changed. (I updated the README to reflect these changes.) 

Moreover, I changed the imports to be relative, as they should be, while importing an object within the same python package. This led to some complications in gather.py where the imports are split (if someone has a better solution I would welcome it), I left a comment to explain in more detail (the comment should be removed before merging ?).

I improved the error handling as briefly discussed in my previous PR, by raising error in place and using try/except blocks. I think the use of utils.evlt should be phased out and replaced with raising the appropriate error and try/except blocks (instead of returning a 0 status variable and checking it in utils.evlt which then calls sys.exit, (btw sys.exit does exit the python shell, raising errors does not.)). I have done this in some cases, but there are still some uses of utils.evlt left.

I also changed the way Study.py handles the cluster_name & cluster_module attributes, this getter/setter method avoids the use the kind of hacky SourceFileLoader. I'm not entirely convinced the getter/setter way is the best but I think it's an improvement.

I've tested this for mysql and sql databases, and I haven't had any problems, however I invite you to test it as well as I realise it is quite a big PR.

If I missed anything or if anything is unclear, please feel free to ask !

### CODE REVIEW

**Comment by loiccoyle** [comment]
As we discussed @luxiaohan, I moved the logger settings to the config.py, this means that unfortunately, the workspace class's logger does not respect the settings in config.py as it runs before config.py is loaded. This was happening with the previous implementation also. 

I readded the cluster_module getter/setter because SourceFileLoader loads a file regardless of whether it is part of package, this was causing problems in submission.py, where the logger would not inherit the package's settings and the imports could not be relative. With the implementation in this PR, if the user wants to use a custom cluster class, the file containing it would have to be in the PYTHONPATH, which is still not ideal.  I changed the README to reflect this.

File: pysixdesk/lib/preprocess.py [hunk 1]
```diff
@@ -327,15 +332,19 @@ def concatenate_files(source, dest):
...
+        logger.error("The input file is missing!")
         sys.exit(1)
     elif num == 2:
         wu_id = args[1]
         db_name = args[2]
         run(wu_id, db_name)
+        sys.exit(0)
     else:
-        print("Too many input arguments!")
+        logger.error("To many input arguments!")
```
Comment (luxiaohan) [inline pysixdesk/lib/preprocess.py]: typo

File: pysixdesk/lib/study.py [hunk 2]
```diff
@@ -655,21 +655,21 @@ def submit(self, typ, trials=5, *args, **kwargs):
         if typ == 0:
             input_path = self.paths['preprocess_in']
             # output_path = self.paths['preprocess_out']
-            # exe = os.path.join(utils.PYSIXDESK_ABSPATH, 'lib', 'preprocess.py')
+            # exe = os.path.join(utils.PYSIXDESK_ABSPATH, 'pysixdesk', 'preprocess.py')
```
Comment (luxiaohan) [inline pysixdesk/lib/study.py]: If these lines are redundant, just remove them

File: pysixdesk/lib/utils.py [hunk 3]
```diff
@@ -43,23 +41,19 @@ def download_output(filenames, dest, zp=True):
     '''Download the requested files to the given destinaion.
     If zp is true, then zip the files before download.
     '''
-    status = False
     if not os.path.isdir(dest):
         os.makedirs(dest, 0o755)
 
     for filename in filenames:
+        if not os.path.isfile(filename):
+            raise FileNotFoundError("The file %s doesn't exist, download failed!" % filename)
```
Comment (luxiaohan) [inline pysixdesk/lib/utils.py]: The same, `check` is called in `preprocess.py` and `sixtrack.py`, avoid to use `raise` here.

**Comment by loiccoyle** [comment]
To answer your question on using logger in a custom cluster, it depends on how the custom logger's cluster is implemented. If the custom cluster is a subclass of the HTCondor cluster then it will would use HTCondor's logger which should work as expected. If it is a completely independent class then the user would have to use a child logger of the pysixdesk package, this would be done by doing in the custom cluster class:
```python
import logging
...
self._logger = logging.getLogger('pysixdesk.customcluster')
```
This will create a logger for the custom cluster which inherits the settings of the pysixdesk package.

Perhaps the better option would be to move HTCondor's logger to the Cluster abstract class, then any custom cluster which subclasses from the Cluster ABC, will also get the correct logger. If it is not clear I can show an example.

**Comment by luxiaohan** [comment]
Yes, it's the better solution to move the HTCondor's logger to the abstract class. The reason we need Cluster ABC is to make sure that the custom cluster have the same interface.
And please do some tests with all these changes, I will approve this PR after your test.
P.S.  you are right, we need an automatic test system. If you can set it up, please just do it! Thanks a lot~

**Comment by loiccoyle** [comment]
I've been testing with sql and mysql, and as far as I can tell, everything is working as expected. I will look into the automatic testing. Thanks for the comments and suggestions!

**Pull request merged**